akka-actor/src/main/scala/akka/util/Unsafe.java

```diff
@@ -23,8 +27,66 @@
             }
             if (found == null) throw new IllegalStateException("Can't find instance of sun.misc.Unsafe");
             else instance = found;
+            stringValueFieldOffset = instance.objectFieldOffset(String.class.getDeclaredField("value"));
+
+            // See Oracle section 1.5.3 at:
+            // https://docs.oracle.com/javase/8/docs/technotes/guides/versioning/spec/versioning2.html
+            final int[] version = Arrays.
+                stream(System.getProperty("java.specification.version").split("\\.")).
+                mapToInt(Integer::parseInt).
+                toArray();
+            javaVersion = version[0] == 1 ? version[1] : version[0];
         } catch (Throwable t) {
             throw new ExceptionInInitializerError(t);
         }
     }
+
+    public static void copyUSAsciiStrToBytes(String str, byte[] bytes) {
+        if (javaVersion > 8) {
+            final byte[] chars = (byte[]) instance.getObject(str, stringValueFieldOffset);
+            System.arraycopy(chars, 0, bytes, 0, chars.length);
+        } else {
+            final char[] chars = (char[]) instance.getObject(str, stringValueFieldOffset);
+            int i = 0;
+            while (i < chars.length) {
+                bytes[i] = (byte) chars[i++];
+            }
+        }
```

Comment:
Another way would be to check the type of `instance.getObject(str, stringValueFieldOffset)` with `instanceof` and select what to do based on that instead of the java version. Should have pretty much equal performance and might be more tolerant for other jvm implementations.

Reply:
`javaVersion` is a constant once calculated which might cause the JIT to remove one branch of the code and leaving a straight call to either Java 8 or 9 branch, that I don't think we will get with `instanceof`

I thought about it but then, being a documented property, is not so bad IMHO.